### ISSUE TITLE
Water drop not visible when pusling

### ISSUE THREAD

**Issue by arouinfar** [phetsims/wave-interference#37]
For #31 

In Unfuddle Ticket [#2535](https://phet.unfuddle.com/a#/projects/9404/tickets/by_number/2535), Wendy Adams reported:
> When I turn the faucet off and press pulse, the wave oscillates but there is no visible drop of water.  At first I saw a drop every once in awhile, now I don’t.  I tried reset all, still did it.  Then I closed the sim and opened a new one and still can’t see the drop if I’m on pulse.  If I switch back to “on” then I see the drop.

I don't think there's any reason why we wouldn't want the water drop to be visible when pulsing the water source. Seems like a legacy bug, so assigning @samreid so he's aware.

**Comment by samreid** [comment]
I'm putting this on hold until we have a decision about https://github.com/phetsims/wave-interference/issues/15#issuecomment-410761082

**Comment by samreid** [comment]
In #15 we decided we wanted the wave to start immediately when the button is pressed.  If we use that same idea for pulse mode, then no water drop would be shown because the wave would already start when the button is pressed.  Or worse (which is the current behavior on phettest), the pulse shows drops falling that disappear before they hit the water.  @arouinfar can you please review and advise?

**Comment by arouinfar** [comment]
> I don't think there's any reason why we wouldn't want the water drop to be visible when pulsing the water source.

I may have to eat my own words! The current behavior is master certainly looks problematic. Perhaps the solution would be to hide the water drop in pulse mode? Let's discuss at design meeting.

**Comment by samreid** [comment]
> Perhaps the solution would be to hide the water drop in pulse mode?

That would be easy to implement, and wouldn't tamper with the rest of our design for phase.  But it does sound buggy to show a water faucet above the water and waves appearing in the water without anything between them.  How about for water/pulse we put the hose at the same y coordinate as the wave source (not vertically offset), and not show any drops?

**Comment by samreid** [comment]
@kathy-phet says: the current behavior is great for continuous mode.  For pulse mode, there could be a temporal delay so it only shows the wave when the water starts.

**Comment by samreid** [comment]
The current implementation follows this approach:

0. The user starts the wave
1. The model drives the lattice to oscillate immediately
2. The wave values make water drops appear above the water (according to the phase), purely in the view

This was a shortcut I took to try to get things working with minimal effort.  It seems this strategy will be insufficient to handle pulse mode or side view properly (as far as I know).  Perhaps we will need to model the water drops in the model like so:
1. The user starts the faucet
2. The model creates a water drop
3. The water drop falls
4. The collision with the water drop and the water starts the water oscillating

If we are going to implement this strategy (with the delay), then it would be nice to be consistent with it and use it for continuous mode too.  However, @kathy-phet said the current behavior (without delay) is great for continuous mode (see preceding comment).  If we implement this new model-based strategy and want to maintain the prior behavior, maybe we could create water drops in the air in the model (as is currently done for the view-based approach).

@kathy-phet and @arouinfar do you think it will work for me to implement the model-based approach (with the time delay) so we'll have that for pulse mode and side view--then we can evaluate whether it is reasonable for continuous mode too?

**Comment by samreid** [comment]
I agree with the plan of "let's try it and see" but while it's on my mind I wanted to jot down that you'll see frequency and amplitude changes immediately in the new drops that fall out.  We can also change the speed of the drops--slower speed means longer delay but you'll see more drops in the air at once.  Faster falling drops means less delay to when the wave pool reacts.

**Comment by samreid** [comment]
> It all depends on how quickly the water wave responds to increase in amplitude and frequency changes in continuous mode. If it is a relatively quick response, it could work, but if having the delay creates a delay such that it deters easy cause - effect it would be a problem. It will likely be quick enough response, but hard to know until I play with it.

The design team reviewed this on Thursday and we agreed that the response time was quick enough that we could use the water drop model.

**Comment by samreid** [comment]
I've finished the initial draft of water drops, such that the water drop stores a frequency, amplitude and position for the lattice oscillator that is applied when the drop hits.  I'm aware of two known problems, which are noted in the two issues linked before this comment.  @arouinfar can you please review this issue and make recommendations for how to proceed, or close this issue if the only remaining work is already described in those other issues?
